clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -2446,37 +2376,41 @@ bool SemaHLSL::CheckBuiltinFunctionCall(unsigned BuiltinID, CallExpr *TheCall) {
   case Builtin::BI__builtin_hlsl_any: {
     if (SemaRef.checkArgCount(TheCall, 1))
       return true;
+    if (CheckAnyScalarOrVector(&SemaRef, TheCall, 0))
+      return true;
     break;
   }
   case Builtin::BI__builtin_hlsl_asdouble: {
     if (SemaRef.checkArgCount(TheCall, 2))
       return true;
-    if (CheckUnsignedIntRepresentation(&SemaRef, TheCall))
+    if (CheckScalarOrVector(&SemaRef, TheCall, SemaRef.Context.UnsignedIntTy,
+                            0)) // only check for uint
+      return true;
+    if (CheckScalarOrVector(&SemaRef, TheCall, SemaRef.Context.UnsignedIntTy,
+                            1)) // only check for uint
```

Comment:
```suggestion
    if (CheckScalarOrVector(&SemaRef, TheCall, 
                            /*only check for uint*/ SemaRef.Context.UnsignedIntTy,
                            1))
```

Reply:
or remove the comment about line 2386 so you aren't duplicating the comment on line 2387 and 2390